Cron/Images.php

```diff
@@ -0,0 +1,236 @@
+     */
+    private $productCollection;
+    /**
+     * @var ProductModel
+     */
+    private $product;
+    /**
+     * @var UrlHelper
+     */
+    private $urlHelper;
+    /**
+     * Directory List
+     *
+     * @var DirectoryList
+     */
+    protected $directoryList;
+
+    /**
+     * File interface
+     *
+     * @var File
+     */
+    protected $file;
+    /**
+     * Images constructor.
+     * @param LoggerInterface $logger
+     * @param ObjectManagerInterface $objectManager
+     * @param StockbaseClientFactory $stockbaseClientFactory
+     * @param StockbaseConfiguration $config
+     * @param ProductCollection $productCollection
+     * @param ProductModel $product
+     * @param UrlHelper $urlHelper
+     * @param DirectoryList $directoryList
+     * @param File $file
+     */
+    public function __construct(
+        LoggerInterface $logger,
+        ObjectManagerInterface $objectManager,
+        StockbaseClientFactory $stockbaseClientFactory,
+        StockbaseConfiguration $config,
+        ProductCollection $productCollection,
+        ProductModel $product,
+        UrlHelper $urlHelper,
+        DirectoryList $directoryList,
+        File $file
+    ) {
+        $this->logger = $logger;
+        $this->stockbaseClientFactory = $stockbaseClientFactory;
+        $this->objectManager = $objectManager;
+        $this->config = $config;
+        $this->productCollection = $productCollection;
+        $this->product = $product;
+        $this->urlHelper = $urlHelper;
+        $this->directoryList = $directoryList;
+        $this->file = $file;
+    }
+
+    /**
+     * Executes the job.
+     */
+    public function execute()
+    {
+        // validate configuration:
+        if (!$this->config->isModuleEnabled() || !$this->config->isImagesSyncEnabled()) {
+            return;
+        }
+        // start process:
+        $this->logger->info('Synchronizing Stockbase images...');
+        // get all the eans:
+        $allEans = $this->getAllEans();
+        // get processed eans:
+        $processedEans = json_decode($this->getImagesEans()) ?: array();
+        // process 100 unprocessed eans at a time:
+        $eans = array_slice(array_diff($allEans, $processedEans), 0, 100);
+        try {
+            $client = $this->stockbaseClientFactory->create();
+            $images = $client->getImages($eans);
+        } catch (Exception $e) {
+            $this->logger->info('Cron runImageImport error: '.$e->getMessage());
+            return false;
+        }
+        // download and save the images locally:
+        $this->saveImageForProduct($images);
+        // update the processed images configuration:
+        $processedEans = array_merge($processedEans, $eans);
+        $encodedEans = json_encode($processedEans);
+        $this->saveImagesEans($encodedEans);
+        $this->logger->info('Stockbase images synchronization complete.');
+    }
+
+    /**
+     * @return array
+     */
+    public function getAllEans()
```

Comment:
I think we can make this more efficient, let's investigate that.

Reply:
Yes, this is just the first commit, here we based the code on the feature from the M1 version (which I also can be improved).